coreservices/onboarding/src/CatenaX.NetworkServices.Onboarding.Service/Controllers/OnboardingController.cs

```diff
@@ -34,172 +33,117 @@ public OnboardingController(ILogger<OnboardingController> logger, IHttpClientFac
         [HttpGet]
         [Route("company/{realm}/{oneId}")]
         [ProducesResponseType(typeof(Company), (int)HttpStatusCode.OK)]
-        public async Task<IActionResult> GetOneObject([FromRoute] string realm, [FromRoute] string oneId, [FromHeader] string authorization)
+        public Task<IActionResult> GetOneObject([FromRoute] string realm, [FromRoute] string oneId, [FromHeader] string authorization)
         {
-
-            if (!ValidateTokenAsync(realm, authorization, out var response))
-            {
-                return response;
-            }
-            else
-            {
+            return ValidateTokenAsync(realm, authorization, async () => {
                 var result = await _onboardingBusinessLogic.GetCompanyByOneId(oneId);
                 return new OkObjectResult(result);
-            }
+            });
         }
 
         [HttpPost]
         [Route("company/{realm}/users")]
-        public async Task<IActionResult> CreateUsers([FromRoute] string realm, [FromHeader] string authorization, [FromBody] List<User> userToCreate)
+        public Task<IActionResult> CreateUsers([FromRoute] string realm, [FromHeader] string authorization, [FromBody] List<User> userToCreate)
```

Comment:
please keep the tasks async and await the ValidateTokenAsync

so: return await ValidateTokenAsync

Reply:
changed as discussed. Renamed asynchronous methods to '*Async' + use of lambda-syntax.
BTW: in your implementation ValidateTokenAsync was actually implemented waiting synchronously (using Task.Result instead of await). That synchronous wait on Result wouldn't magically disappear by adding the keyword async to the calling method as this would just cause generation of continuations (.ContinueWith...) only for those places where await is being used. It doesn't know about the synchronous waiting for the http-call inside ValidateTokenAsync.
That is fixed now ;-)